src/routing/approved.rs

```diff
@@ -1678,6 +1696,12 @@ impl Approved {
                 commands.extend(self.send_sync(self.section.clone(), self.network.clone())?);
             }
 
+            let sibling_key = if new_prefix != old_prefix {
+                self.section_key(&new_prefix.sibling()).copied()
```

Comment:
at this momenet, the section_key of the sibling shall be presented, right?

Reply:
Yes, it should be `Some` if that's what you are asking. It's tricky to enforce that statically though.